docs/development/secrets_management.md

```diff
@@ -86,20 +86,16 @@ This is to ensure a smooth exchange of certificate during a CA rotation (typical
   - In phase 1, servers still use their old/existing certificates to allow clients to update their CA bundle used for verification of the servers' certificates.
   - In phase 2, the old CA is dropped, hence servers need to get a certificate signed by the new/current CA. At this point in time, clients have already adapted their CA bundles.
 
-#### Always Sign Server Certificates with Current CA
-
-In case you control all clients and update them at the same time as the server, it is possible to make the secrets manager generate even server certificates with the new/current CA.
-This can help to prevent certificate mismatches when the CA bundle is already exchanged while the server still serves with a certificate signed by a CA no longer part of the bundle.
+> :bulb: **Sign Server Certificates with Current CA, in case you control all clients and update them at the same time as the server**. It is possible to make the secrets manager generate even server certificates with the new/current CA.
+> This can help to prevent certificate mismatches when the CA bundle is already exchanged while the server still serves with a certificate signed by a CA no longer part of the bundle.
 
 Let's consider the two following examples:
 
 1. `gardenlet` deploys a webhook server (`gardener-resource-manager`) and a corresponding `MutatingWebhookConfiguration` at the same time. In this case, the server certificate should be generated with the new/current CA to avoid above mentioned certificate mismatches during a CA rotation.
 2. `gardenlet` deploys a server (`etcd`) in one step, and a client (`kube-apiserver`) in a subsequent step. In this case, the default behaviour should apply (server certificate should be signed by old/existing CA).
 
-#### Always Sign Client Certificate with Old CA
```

Comment:
Can we keep the subheadings (feel free to rephrase)? I think this will lead to a more readable structure.

Reply:
I do not think the headings make it more clear and readable on this level. To me it just look like a statement in bold font on single line which is why I opened #8183.

Now the examples are also more clear are on the `Certificate Signing` level, not the just covering the exception cases.

old vs. new 
![image](https://github.com/gardener/gardener/assets/6798531/56333db7-8cbf-4e7a-a269-ff4fd65efb7c)


![image](https://github.com/gardener/gardener/assets/6798531/0728e47a-e81a-4daf-9aa8-bbb480b1c909)

Reply:
I agree with @timuthy that removal of the subheadings, additional indentation and the grey font color side-effect are not necessarily improving reading experience.

However, I also agree with @vpnachev that this section can be improved. From my perspective, there are three different sections, i.e. default, exception 1 and exception 2.

How about the the following:

```
### Certificate Signing

#### Default Behaviour

By default, client certificates are signed by the current CA while server certificate are signed by the old CA (if it exists).
This is to ensure a smooth exchange of certificate during a CA rotation (typically has two phases, ref [GEP-18](../proposals/18-shoot-CA-rotation.md#rotation-sequence-for-cluster-and-client-ca)):

- Client certificates:
  - In phase 1, clients get new certificates as soon as possible to ensure that all clients have been adapted before phase 2.
  - In phase 2, the respective server drops accepting certificates signed by the old CA.
- Server certificates:
  - In phase 1, servers still use their old/existing certificates to allow clients to update their CA bundle used for verification of the servers' certificates.
  - In phase 2, the old CA is dropped, hence servers need to get a certificate signed by the new/current CA. At this point in time, clients have already adapted their CA bundles.

#### Alternative: Sign Server Certificates with Current CA

In case you control all clients and update them at the same time as the server, it is possible to make the secrets manager generate even server certificates with the new/current CA.
This can help to prevent certificate mismatches when the CA bundle is already exchanged while the server still serves with a certificate signed by a CA no longer part of the bundle.

Let's consider the two following examples:

1. `gardenlet` deploys a webhook server (`gardener-resource-manager`) and a corresponding `MutatingWebhookConfiguration` at the same time. In this case, the server certificate should be generated with the new/current CA to avoid above mentioned certificate mismatches during a CA rotation.
2. `gardenlet` deploys a server (`etcd`) in one step, and a client (`kube-apiserver`) in a subsequent step. In this case, the default behaviour should apply (server certificate should be signed by old/existing CA).

#### Alternative: Sign Client Certificate with Old CA

In the unusual case where the client is deployed before the server, it might be useful to always use the old CA for signing the client's certificate.
This can help to prevent certificate mismatches when the client already gets a new certificate while the server still only accepts certificates signed by the old CA.

Let's consider the two following examples:

1. `gardenlet` deploys the `kube-apiserver` before the `kubelet`. However, the `kube-apiserver` has a client certificate signed by the `ca-kubelet` in order to communicate with it (e.g., when retrieving logs or forwarding ports). In this case, the client certificate should be generated with the old CA to avoid above mentioned certificate mismatches during a CA rotation.
2. `gardenlet` deploys a server (`etcd`) in one step, and a client (`kube-apiserver`) in a subsequent step. In this case, the default behaviour should apply (client certificate should be signed by new/current CA).
```

Changes:
- Drop "always" three times as there are obviously exceptions/alternatives
- Separate all three sections, i.e. add a leading "default" section
- Add a second example to the last examples block so that it is analogous to the previous one and describes both cases (default and alternative)

(An alternative approach would be to have examples in each section, not just in the last two.)